spatialdata/_core/_spatialdata.py

```diff
@@ -425,40 +426,78 @@ def add_image(
+                    "elements used for backing. If you would like to support this use case please leave a comment on "
+                    "https://github.com/scverse/spatialdata/pull/138"
+                )
+            self._add_image_in_memory(name=name, image=image, overwrite=overwrite)
+            # old code to support overwriting the backing file
+            # with tempfile.TemporaryDirectory() as tmpdir:
+            #     store = parse_url(Path(tmpdir) / "data.zarr", mode="w").store
+            #     root = zarr.group(store=store)
+            #     write_image(
+            #         image=self.images[name],
+            #         group=root,
+            #         name=name,
+            #         storage_options=storage_options,
+            #     )
+            #     src_element_path = Path(store.path) / name
+            #     assert isinstance(self.path, str)
+            #     tgt_element_path = Path(self.path) / "images" / name
+            #     if os.path.isdir(tgt_element_path) and overwrite:
+            #         element_store = parse_url(tgt_element_path, mode="w").store
+            #         _ = zarr.group(store=element_store, overwrite=True)
+            #         element_store.close()
+            #     pathlib.Path(tgt_element_path).mkdir(parents=True, exist_ok=True)
+            #     for file in os.listdir(str(src_element_path)):
+            #         src_file = src_element_path / file
+            #         tgt_file = tgt_element_path / file
+            #         os.rename(src_file, tgt_file)
+            # from spatialdata._io.read import _read_multiscale
+            #
+            # # reload the image from the Zarr storage so that now the element is lazy loaded, and most importantly,
+            # # from the correct storage
+            # image = _read_multiscale(str(tgt_element_path), raster_type="image")
+            # self._add_image_in_memory(name=name, image=image, overwrite=True)
             elem_group = self._init_add_element(name=name, element_type="images", overwrite=overwrite)
             write_image(
                 image=self.images[name],
                 group=elem_group,
                 name=name,
                 storage_options=storage_options,
             )
-            # reload the image from the Zarr storage so that now the element is lazy loaded, and most importantly,
-            # from the correct storage
             from spatialdata._io.read import _read_multiscale
 
+            # reload the image from the Zarr storage so that now the element is lazy loaded, and most importantly,
+            # from the correct storage
             assert elem_group.path == "images"
-            path = os.path.join(elem_group.store.path, "images", name)
-            image = _read_multiscale(path, raster_type="image")
+            path = Path(elem_group.store.path) / "images" / name
+            image = _read_multiscale(str(path), raster_type="image")
```

Comment:
this is nitpick I undersatnd, but it looks like this is because `ZarrLocation` from `ome-zarr-py` doesn't accept paths but only strings? would it be a solution to put the str casting only there, and stick to `Path()` in the repo? instead of os.path.join, used also above

Reply:
Actually `ome-zarr-py` accepts `Path`, I was using an `assert` that was not necessary. I changed this and now I have removed the conversion to `str`. I am still keeping `SpatialData.path`as a `str` (we can change it later)